dd-java-agent/agent-otel/otel-shim/src/main/java/datadog/opentelemetry/shim/context/OtelContext.java

```diff
@@ -1,170 +1,106 @@
-      lastPropagated.remove();
-      AgentScope agentScope = ((OtelSpan) this.currentSpan).activate();
-      return new OtelScope(scope, agentScope, this.entries);
-    } else {
-      // propagated context not on the scope stack, capture it here
-      lastPropagated.set(this);
-      return new Scope() {
-        @Override
-        public void close() {
-          lastPropagated.remove();
-          scope.close();
-        }
-      };
-    }
-  }
-
-  public static Context current() {
-    // Check for propagated context not on the scope stack
-    Context context = lastPropagated.get();
-    if (null != context) {
-      return context;
-    }
-    // Check empty context
-    AgentScope agentCurrentScope = AgentTracer.activeScope();
-    if (null == agentCurrentScope) {
-      return OtelContext.ROOT;
-    }
-    // Get OTel current span
-    Span otelCurrentSpan = null;
-    AgentSpan agentCurrentSpan = agentCurrentScope.span();
-    if (agentCurrentSpan instanceof AttachableWrapper) {
-      Object wrapper = ((AttachableWrapper) agentCurrentSpan).getWrapper();
-      if (wrapper instanceof OtelSpan) {
-        otelCurrentSpan = (OtelSpan) wrapper;
-      }
-    }
-    if (otelCurrentSpan == null) {
-      otelCurrentSpan = new OtelSpan(agentCurrentSpan);
+  public boolean equals(Object o) {
+    if (o == null || getClass() != o.getClass()) {
+      return false;
     }
-    // Get OTel root span
-    Span otelRootSpan = null;
-    AgentSpan agentRootSpan = agentCurrentSpan.getLocalRootSpan();
-    if (agentRootSpan instanceof AttachableWrapper) {
-      Object wrapper = ((AttachableWrapper) agentRootSpan).getWrapper();
-      if (wrapper instanceof OtelSpan) {
-        otelRootSpan = (OtelSpan) wrapper;
-      }
-    }
-    if (otelRootSpan == null) {
-      otelRootSpan = new OtelSpan(agentRootSpan);
-    }
-    // Get OTel custom context entries
-    Object[] contextEntries = NO_ENTRIES;
-    if (agentCurrentScope instanceof AttachableWrapper) {
-      Object wrapper = ((AttachableWrapper) agentCurrentScope).getWrapper();
-      if (wrapper instanceof OtelScope) {
-        contextEntries = ((OtelScope) wrapper).contextEntries();
-      }
-    }
-    return new OtelContext(otelCurrentSpan, otelRootSpan, contextEntries);
+    return delegate.equals(((OtelContext) o).delegate);
   }
 
-  /** Last propagated context not on the scope stack; {@code null} if there's no such context. */
-  @Nullable
-  public static Context lastPropagated() {
-    return lastPropagated.get();
+  @Override
+  public int hashCode() {
+    return delegate.hashCode();
   }
 
   @Override
   public String toString() {
-    return "OtelContext{"
-        + "currentSpan="
-        + this.currentSpan.getSpanContext()
-        + ", rootSpan="
-        + this.rootSpan.getSpanContext()
-        + '}';
+    return "OtelContext{" + "delegate=" + delegate + '}';
+  }
+
+  static datadog.context.ContextKey datadogKey(ContextKey key) {
```

Comment:
Can't find usage outside this class. Maybe with some later refactoring? 🤔 
```suggestion
  private static datadog.context.ContextKey datadogKey(ContextKey key) {
```